src/app/services/User.service.ts

```diff
@@ -40,4 +41,17 @@ export class UserService {
     const savedUser = await this.repository.save(userToSave);
     return this.userMap.toDTO(savedUser);
   }
+
+  async updateSourcesOfInterest(token: string, sourcesOfInterest: Source[]) {
+    const claims = this.encryptionService.verify<UserTokenClaims>(token);
+    const user = await this.repository.getById(claims.id);
+
+    if (!user) {
+      throw new NoSuchElementException('user not found');
+    }
```

Comment:
Checking for a user is dubious given the fact that the request has passed jwt validation. A token cannot be generation for a usor that does not exist.

Reply:
this method could theoretically be called as part of another method, maybe coming from a public route.